### PR TITLE
Federate sticky flag dessalines

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: ui/src/api_tests/api.spec.ts [hunk 1]
```diff
@@ -353,6 +375,76 @@ describe('main', () => {
...
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: wrapper(commentForm),
+        }
+      ).then(d => d.json());
+
+      expect(createResponse.comment).toBeUndefined();
```
Comment (Nutomic) [inline ui/src/api_tests/api.spec.ts:426]: Does this not send a proper error message?

**Comment by Nutomic** [comment]
All the sticky tests are interacting with the alpha instance, so this doesnt really test whether it federates. I would try the comment creation from another instance instead. Also it would be better to make a separate test, so its easier to see what exactly failed.

File: ui/src/api_tests/api.spec.ts (hunk 1 above)
Comment (dessalines) [reply ui/src/api_tests/api.spec.ts:426]: Good call, looks like it does return `{ error: 'locked' }`. I'll add this instead.

**Comment by dessalines** [comment]
> All the sticky tests are interacting with the alpha instance, so this doesnt really test whether it federates. 

It does test both federated locked and stickied, it fetches from B. https://github.com/LemmyNet/lemmy/pull/1021/files#diff-eef481bbcdee103e228e69331178b559R401

You're right tho that this doesn't test if B tries to make a comment on a locked post on A.... that comes later with all the API related checks we have to add to the activitypub receives. That's for another ticket tho.

**Pull request merged**